File: snow-chat-flutter/lib/models/upload_result.dart
/// 文件上传结果数据类。
///
/// 对应后端 `UploadResult` record：
/// - [key]：对象存储中的 key（如 `avatars/uuid.jpg`），用于持久化到用户 avatar 字段；
/// - [url]：访问该对象的 pre-signed URL（带有效期），前端可直接用于展示。
///
/// 使用 `fromJson` 工厂方法从后端响应的 `data` 字段解析。
class UploadResult {
  /// 对象 key（如 `avatars/uuid.jpg`）
  final String key;

  /// pre-signed 访问 URL（带有效期）
  final String url;

  const UploadResult({required this.key, required this.url});

  /// 从后端 JSON 响应的 data 字段构造 UploadResult。
  /// 若 key 或 url 缺失，返回 null（调用方应做空判断）。
  factory UploadResult.fromJson(Map<String, dynamic> json) {
    return UploadResult(
      key: json['key'] as String? ?? '',
      url: json['url'] as String? ?? '',
    );
  }
}

File: snow-chat-im-backend/snow-im-api/src/main/java/com/stylesmile/chat/dto/UploadResult.java
package com.stylesmile.chat.dto;

import io.swagger.v3.oas.annotations.media.Schema;

/**
 * 文件上传结果DTO
 *
 * <p>采用"私有 + Pre-signed URL"策略：
 * <ul>
 *   <li>{@code key}：对象存储中的 key（如 {@code avatars/uuid.jpg}），用于持久化到数据库；</li>
 *   <li>{@code url}：访问该对象用的 pre-signed URL（带有效期），前端可直接用于展示。</li>
 * </ul>
 *
 * <p>使用 Java 17 record 简化不可变数据载体定义，自动生成构造器、访问器、equals/hashCode/toString。
 *
 * @author mmm
 */
@Schema(description = "文件上传结果")
public record UploadResult(
        @Schema(description = "对象存储key", example = "avatars/abc123.jpg")
        String key,
        @Schema(description = "访问URL（pre-signed或永久）", example = "https://cdn.example.com/avatars/abc123.jpg")
        String url
) {}
